### PR TITLE
Authentication section

### PR DESCRIPTION
**Pull request by Ariesleo**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
   <meta name="viewport" content="width=device-width,initial-scale=1.0">
   <link rel="icon" href="<%= BASE_URL %>favicon.ico">
   <title>Transcribe</title>
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (shakyaprasen) [inline public/index.html:10]: stylesheet can be imported in app.vue file or main file

File: src/components/TextBox.vue [hunk 2]
```diff
@@ -12,10 +12,10 @@
   </form>
   <div class="p-2">
     <b-button type="button" class="sbutton-imp" @click="addItem()" variant="light">
-      Save
+      <b-icon  icon="collection-play"></b-icon>Save
     </b-button>
   </div>
-    <b-alert
+    <b-alert class="mx-auto m-2" style="width:40%"
```
Comment (shakyaprasen) [inline src/components/TextBox.vue:18]: you can use class instead of stylke here

File: src/components/TextBox.vue [hunk 3]
```diff
@@ -30,6 +30,7 @@ import { db } from '../firebase/db';
 
 export default {
   name: 'TextBox',
+  props: ['info'],
```
Comment (shakyaprasen) [inline src/components/TextBox.vue:33]: use detailed props declaration with validation and default value if required

File: src/components/navBar.vue [hunk 4]
```diff
@@ -28,7 +30,36 @@
...
 <script>
+import * as firebase from 'firebase/app';
+import 'firebase/auth';
+
 export default {
   name: 'NavBar',
+  created() {
+    firebase.auth().onAuthStateChanged((val) => {
+      // this.loggedIn = !!user;
+      if (val) {
```
Comment (shakyaprasen) [inline src/components/navBar.vue:41]: this could be written as `this.loggedIn = !!val`

File: src/components/topButtons.vue [hunk 5]
```diff
@@ -48,22 +41,25 @@ export default {
...
-      this.isYoutubeVisible = true;
-      this.youtubeSrc = this.youtubeInputLink;
-    },
     videoClick(event) {
       this.isVideoVisible = true;
       const vm = this;
       const reader = new FileReader();
       reader.onload = (e) => {
         vm.videoSrc = e.target.result;
+        this.$refs.Match.addEventListener('loadedmetadata', () => {
```
Comment (shakyaprasen) [inline src/components/topButtons.vue:50]: check to see if this.$refs.Match exists before adding an event listener

File: public/index.html (hunk 1 above)
Comment (shakyaprasen) [reply public/index.html:10]: normally we don't need to change the index file.

File: src/views/Files.vue [hunk 6]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <div class="Files">
+    <NavBar />
+      <table class="table table-borderless mx-auto m-2 text-center" style="width:40%">
```
Comment (shakyaprasen) [inline src/views/Files.vue:4]: style can be moved to class

File: src/views/Register.vue [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    return {
+      email: '',
+      password: '',
+      error: '',
+    };
+  },
+  methods: {
+    async registerNew() {
+      try {
+        if (this.email !== '' && this.password !== '') {
```
Comment (shakyaprasen) [inline src/views/Register.vue:67]: we will need better front end validation later. Don't forget this later

File: src/views/SignIn.vue [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+    return {
+      email: '',
+      password: '',
+      error: '',
+    };
+  },
+  methods: {
+    async SingIn() {
+      try {
+        if (this.email !== '' && this.password !== '') {
```
Comment (shakyaprasen) [inline src/views/SignIn.vue:80]: same with validation here. Better validation for later sprints

File: src/views/Transcribe.vue [hunk 9]
```diff
@@ -1,23 +1,34 @@
...
 <script>
 // @ is an alias to /src
 import NavBar from '@/components/NavBar.vue';
-import TopButtons from '@/components/TopButtons.vue';
+// import TopButtons from '@/components/TopButtons.vue';
 import TextBox from '@/components/TextBox.vue';
 
 export default {
   name: 'Transcribe',
+  props: ['info'],
```
Comment (shakyaprasen) [inline src/views/Transcribe.vue:17]: better prop declarations and validation `https://vuejs.org/v2/guide/components-props.html#Prop-Validation`

File: src/views/Transcriptions.vue [hunk 10]
```diff
@@ -1,40 +1,45 @@
...
+              @click="deleteItem(transDataItem.id)">
+                <b-icon icon="Trash"></b-icon>
+              </button>
+            </td>
+
+          </tr>
+        </tbody>
+      </table>
+      <a href="/#/transcribe" class="btn btn-info m-1" role="button">Add New</a>
+      <b-alert class="mx-auto m-2" style="width:50%"
```
Comment (shakyaprasen) [inline src/views/Transcriptions.vue:29]: can style be moved to class or does it not override the default width?

File: src/views/Upload.vue [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+            <textarea class="form-control p-2" rows="3" id="comments"
+            placeholder="Comments" v-model="newComment"></textarea>
+        </div>
+    </form>
+    <div class="p-2">
+        <b-button type="button" class="sbutton-imp" @click="addItem()" variant="light">
+        Save
+        </b-button>
+    </div>
+    <b-alert class="mx-auto m-2" style="width:50%"
```
Comment (shakyaprasen) [inline src/views/Upload.vue:24]: same with styule here

File: src/views/Upload.vue [hunk 12]
```diff
@@ -0,0 +1,88 @@
+<template>
+  <div class="upload">
+    <NavBar />
+    <TopButtons @durationdata='durationdatavalue' />
+    <form class="mx-auto p-2" style="width:50%">
```
Comment (shakyaprasen) [inline src/views/Upload.vue:5]: style here same.

File: src/views/Upload.vue [hunk 13]
```diff
@@ -0,0 +1,88 @@
+<template>
+  <div class="upload">
+    <NavBar />
+    <TopButtons @durationdata='durationdatavalue' />
```
Comment (shakyaprasen) [inline src/views/Upload.vue:4]: can you use camelCase for events and function name i.e. @durationData='durationDataValue'

**Review by shakyaprasen — APPROVED** [review]
Merge after addressing the comments

File: src/views/Files.vue (hunk 6 above)
Comment (Ariesleo) [reply src/views/Files.vue:4]: done

**Pull request merged**